kafka-connect-s3/src/main/java/io/confluent/connect/s3/storage/S3Storage.java

```diff
@@ -75,25 +74,31 @@ public S3Storage(S3SinkConnectorConfig conf, String url) {
    */
   public AmazonS3 newS3Client(S3SinkConnectorConfig config) {
     ClientConfiguration clientConfiguration = newClientConfiguration(config);
+
     AmazonS3ClientBuilder builder = AmazonS3ClientBuilder.standard()
-                                        .withAccelerateModeEnabled(
-                                            config.getBoolean(WAN_MODE_CONFIG)
-                                        )
-                                        .withPathStyleAccessEnabled(true)
-                                        .withCredentials(config.getCredentialsProvider())
-                                        .withClientConfiguration(clientConfiguration);
+            .withAccelerateModeEnabled(config.getBoolean(WAN_MODE_CONFIG))
+            .withPathStyleAccessEnabled(true)
+            .withClientConfiguration(clientConfiguration);
+
+    String accessKey = config.getString(S3_ACCESS_KEY);
+    String secretKey = config.getString(S3_SECRET_KEY);
+    if (accessKey != null && secretKey != null) {
+      BasicAWSCredentials awsCreds = new BasicAWSCredentials(accessKey, secretKey);
+      builder.withCredentials(new AWSStaticCredentialsProvider(awsCreds));
+    } else {
+      builder.withCredentials(config.getCredentialsProvider());
+    }
 
     String region = config.getString(REGION_CONFIG);
     if (StringUtils.isBlank(url)) {
       builder = "us-east-1".equals(region)
-                ? builder.withRegion(Regions.US_EAST_1)
```

Comment:
Let's avoid stylistic changes in this PR. Will make it slightly easier to backport too.

Reply:
rookie mistake, fixed.